plugins/BEdita/Core/src/Model/Entity/ObjectEntity.php

```diff
@@ -185,6 +185,33 @@ protected function getRelationships()
                 ],
                 true
             );
+            $entityDest = $this->getTable()->associations()->getByProperty($relationship)->getTarget()->newEntity();
+            if ($entityDest instanceof JsonApiSerializable) {
+                $destObj = $entityDest->jsonApiSerialize(JsonApiSerializable::JSONAPIOPT_BASIC);
+                if (empty($destObj['type'])) {
+                    $options = ['name' => $relationship];
+                    $result = TableRegistry::get('ObjectTypes')
+                        ->find('byRelation', $options)
+                        ->find('list')
+                        ->toArray();
+                    $available = Router::url(
+                        [
+                            '_name' => 'api:objects:index',
+                            'object_type' => 'objects',
+                            'filter' => ['type' => array_values($result)],
+                        ],
+                        true
+                    );
+                } else {
+                    $available = Router::url(
```

Comment:
These lines are not covered by unit tests. It should be possible to cover them by adding a case to the existing test case's data provider that asserts the presence of the `relationships.roles.links.available` key for a User.

Reply:
It seems that test is already covered by `BEdita\API\Test\TestCase\Controller\ObjectsControllerTest->testIndex (/var/www/workspace/bedita4/plugins/BEdita/API/tests/TestCase/Controller/ObjectsControllerTest.php:363)`

Reply:
@didoda Yes, but that test case is not specifically covering these lines. Instead of extending that test case coverage, I'd duplicate that test in `ObjectEntityTest` class: code duplication is fine within tests. 🙃

Reply:
I just put a breakpoint in that line and test passed from there... covering means something else than "passing from there", doesn't it? Then I'll write a test method in `ObjectEntityTest`.

Reply:
Code coverage is measured only for classes and/or methods explicitly set using `@covers` annotation. This helps preventing unintentionally covered code ("coverage by accident"), where a test case that's supposed to test the functionality of a component A does indeed cover code of another component B, thereby not pointing out the need for specific tests for component B.

As a rule of thumb, a `MyWhateverTest` class **SHOULD** only `@covers` methods from `MyWhatever` class in order to avoid confusion, while integration tests should `@coversNothing` since they are "high level" tests that are more focused on the correct interaction between several components rather than each line of code from a single component.

In this case, `ObjectsControllerTest` does only `@covers` methods from `ObjectsController`, therefore lines of code executed in other components (including `ObjectEntity`) are discarded, as those lines are "used"<sup>1</sup> rather than "covered".

See [specifying covered methods](https://phpunit.de/manual/current/en/code-coverage-analysis.html#code-coverage-analysis.specifying-covered-methods) in PHPUnit docs for reference.

------------

<sup>1</sup> "Used" components **SHOULD** be always explicitly stated via the `@uses` annotation in every test case. However, since in our case we use so many components in every operation, this is practically unfeasible. We have therefore configured PHPUnit in a way that it does not enforce the use of `@uses` annotation. If we were to use the "strict" mode, tests that execute lines of code from components not declared via that annotation [would be marked as "risky"](https://phpunit.de/manual/current/en/risky-tests.html#risky-tests.unintentionally-covered-code).